proto/streamlit/proto/GitInfo.proto

```diff
@@ -0,0 +1,28 @@
+/**
+* Copyright 2018-2020 Streamlit Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+syntax = "proto3";
+
+// Message used to update page metadata.
+message GitInfo {
+  string repository = 1;
+  string branch = 2;
+  string module = 3;
+  bool is_head_detached = 4;
+  repeated string untracked_files = 5;
+  repeated string uncommitted_files = 6;
+  bool is_ahead = 7;
```

Comment:
Nit: Can we pick a more meaningful name for `is_ahead`? E.g. `is_ahead_of_remote`?